### PR TITLE
Added a stop to Network feature. There is a race condition on the gar…

### PR DESCRIPTION
**Pull request by mchacki**
This fixes a race condition on the new feature (devel only)

The race is the following:

before the call of `queueGarbageCollection` we check if we are shutting down. afterwards we get the mutex.
In the beginShutdown phase we get the mutex and drop the last posted item.
These two can overtake each other.
Within `queueGarbageCollection ` no-one checks for shutdown again (also not used components e.g. Scheduler) but the can still successfully deploy the workItem again.
This last workItem is "leaked" which causes scheduler shutdown to crash on maintainer-mode (non-canceled job)

### CODE REVIEW

**Comment by mchacki** [comment]
Jenkins
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/6014/

**Comment by mchacki** [comment]
Next try, we inherited red tests from devel http://jenkins.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/6030/

**Comment by mchacki** [comment]
Fixed test cleanups (unrelated):
http://jenkins.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/6046/

File: tests/js/client/shell/shell-analyzer-rest-api.js [hunk 1]
```diff
@@ -246,41 +246,48 @@ function testSuite() {
...
-      db._drop(col.name());
+        // delete with force - must succeed
+        result = arango.DELETE("/_api/analyzer/" + name + "?force=true");
+        assertFalse(result.error);
+      } finally {
+        try {
+          db._drop("ulfColTestLinks");
+        } catch (_) {}
+        try {
+          db._drop("ulfViewTestLinks");
```
Comment (neunhoef) [inline tests/js/client/shell/shell-analyzer-rest-api.js]: Does this need a try/catch?

Comment (neunhoef) [reply tests/js/client/shell/shell-analyzer-rest-api.js]: And should it be `db._dropView`?

File: tests/js/common/shell/shell-view-arangosearch.js [hunk 2]
```diff
@@ -43,6 +43,14 @@ var ERRORS = arangodb.errors;
 function ViewSuite () {
   'use strict';
   return {
+    tearDown : function () {
+      try {
+        db._dropView("abc");
+      } catch (_) {}
+      try {
```
Comment (neunhoef) [inline tests/js/common/shell/shell-view-arangosearch.js]: See above: Does this need try/catch?

File: tests/js/client/shell/shell-analyzer-rest-api.js (hunk 1 above)
Comment (mchacki) [inline tests/js/client/shell/shell-analyzer-rest-api.js]:
```suggestion
          db._dropView("ulfViewTestLinks");
```

File: tests/js/common/shell/shell-view-arangosearch.js (hunk 2 above)
Comment (mchacki) [reply tests/js/common/shell/shell-view-arangosearch.js]: I think right now it does not throw, but try catch does not make it worse

**Pull request merged**